lib/execution/executor.ex

```diff
@@ -530,28 +440,226 @@ defmodule WaspVM.Executor do
     {frame, Map.put(vm, :stack, Stack.push(stack, result))}
   end
 
-  defp exec_inst({frame, vm}, :i32_rotl) do
+  defp exec_inst({frame, vm}, :i32_rem_u) do
     {[b, a], stack} = Stack.pop_multiple(vm.stack)
-    j2 = b - (32 * Integer.floor_div(b, 32))
-
-    answer = rotl(a, j2)
 
-    {frame, Map.put(vm, :stack, Stack.push(stack, answer))}
+    if b == 0 do
+      {:error, :undefined}
+    else
+      {frame, Map.put(vm, :stack, Stack.push(stack, a - (b*trunc(a/b))))}
```

Comment:
This isn't very readable, lets split this into a few steps, perhaps by piping? Something like

```elixir
c = 
  a
  |> Kernel./(b)
  |> trunc()
  |> Kernel.*(b)

res = a - c

```